### ISSUE TITLE
Alternative to 47 accessor functions for the metadata columns

### ISSUE THREAD

**Issue by patrickbarks** [jonesor/Rcompadre#40]
I'm wondering if there's an alternative to having accessor functions for every single `metadata` column (the `mat`-related ones I like)? A few (fairly minor) downsides I see:

- they pollute the namespace and make `Rcompadre::` less useful for finding functions (I assume the `::` autcomplete is limited to RStudio)

- relatedly, they reduce the ease of using RStudio's column autocomplete options for piped expressions to `subset()`, `mutate()`, etc. (though this feature is currently finicky anyway)

- based on my limited experience, other `S4` classes don't seem to use accessors for columns within a `data` slot (e.g. `sp`)

One alternative (based on the `sp` package) would be to have `CompadreData` methods for `$`, `$<-`, and `names` that directly access the `data` slot, e.g. so `db$SpeciesAuthor` is equivalent to `db@data$SpeciesAuthor`. This enables autocomplete in RStudio using `db$`. Even apart from autocomplete, I think it would be useful in that it gives users quicker access to the good stuff (i.e. everything except `version`).

**Comment by jonesor** [comment]
That does sound simpler, and simple is good... - what do you think @iainmstott ?

**Comment by iainmstott** [comment]
Definitely a good idea! I should have thought of that. It also fits naturally given there's already a [ method (although I still want to implement a [ <- method). 

I'd favour every column of the data except "mat": as one needs matA() etc. to access the specific components of the CompadreM objects, using mat() seems like a natural fit.

**Comment by patrickbarks** [comment]
Ya, I like the `mat`-related accessors. Here's the code I tested out. I'll leave it for Iain or Owen to edit/incorporate, to avoid merge conflicts (remember to update 'CompadreData' to 'CompadreDB' if appropriate).

```
#' @rdname CompadreDataMethods
#' @export
setMethod("$", signature = "CompadreData", 
          function(x, name) {
            if (!("data" %in% slotNames(x))) {
              stop("$ method requires CompadreData object with slot 'data'")
            }
            x@data[[name]]
          }
)

#' @rdname CompadreDataMethods
#' @export
setReplaceMethod("$", signature = "CompadreData", 
                 function(x, name, value) { 
                   if (!("data" %in% slotNames(x))) {
                     stop("$<- method requires CompadreData object with slot 'data'")
                   }
                   x@data[[name]] <- value 
                   x
                 }
)

#' @rdname CompadreDataMethods
#' @export
names.CompadreData <- function(x) {
  if (!("data" %in% slotNames(x))) {
    stop("names method requires CompadreData object with slot 'data'")
  }
  names(x@data)
}
```

**Comment by patrickbarks** [comment]
closed by #43 and subsequent updates
